### DISCUSSION TITLE
body is strange by core.request.get_post_args(ctx)

### DISCUSSION THREAD

**Discussion by kevalin** [apache/apisix#8085] [Q&A]
I request a api by postman. 

Body data:
```json
{ "name": "Tom", "age": 12 }
```

```lua
local body = core.request.get_post_args(ctx) 
type(body) == "table"  -- true
for k, v in pairs(body) do
  core.log.warn("k: ", k)  -- it's my body ( a string json("{\r\n    \"name\": \"Tom\",\r\n    \"age\": 12,\r\n}"), so strange )
  core.log.warn("v: ", v)  -- true (why is it 'true'?)
end
```
I expect k == "name" and v == "Tom", and so on.
But, I must use cjson.decode(k).name to get real data. It's not so cool.
What Happened?

**Comment by tzssangglass** [comment]
you can learn form: https://github.com/apache/apisix/blob/6ae923a12fc1b7f8414b66efc053b19c7b5ddd08/apisix/plugins/wolf-rbac.lua#L332-L348

**Reply by kevalin** [reply]
I see your point:
1. `http_content_type == "application/json"`, I should decode body by myself.
2. others, just do it by `core.request.get_post_args(ctx)`.

😄, Thanks.
